utils/src/lib.rs

```diff
@@ -175,6 +176,27 @@ pub fn get_func(sig: &str) -> Result<Function> {
     Ok(func.clone())
 }
 
+pub async fn get_func_etherscan(
+    function_name: &str,
+    contract: Address,
+    args: Vec<String>,
+    chain: Chain,
+    etherscan_api_key: String,
+) -> Result<Function> {
+    let client = Client::new(chain, etherscan_api_key)?;
+    let abi = client.contract_abi(contract).await?;
+    let funcs = abi.functions.get(function_name).unwrap();
+
+    for func in funcs {
+        let res = encode_args(func, &args);
+        if res.is_ok() {
+            return Ok(func.clone())
+        }
```

Comment:
wonder if we should just be returning the encoded args instead. nbd either, the perf overhead from double encoding should be negl